dicom_utils/validation.py

```diff
@@ -0,0 +1,398 @@
+    ValidationTarget(Tag.Rows, ValidationLevel.VALID, validator=min_max_validator(min=1)),
+    ValidationTarget(Tag.Columns, ValidationLevel.VALID, validator=min_max_validator(min=1)),
+    ValidationTarget(Tag.PixelData, ValidationLevel.VALID, Priority.HIGH, validator=pixel_data_validator),
+    (
+        ValidationTarget(Tag.Laterality, ValidationLevel.VALID, Priority.HIGH),
+        ValidationTarget(Tag.ImageLaterality, ValidationLevel.VALID, Priority.HIGH),
+    ),
+    ValidationTarget(Tag.ViewPosition, ValidationLevel.VALID, Priority.HIGH),
+    ValidationTarget(Tag.PatientSex, ValidationLevel.VALID),
+    ValidationTarget(Tag.EthnicGroup, ValidationLevel.VALID),
+    ValidationTarget(Tag.BreastImplantPresent, ValidationLevel.VALID, Priority.HIGH),
+    ValidationTarget(Tag.PhotometricInterpretation, ValidationLevel.VALID, Priority.HIGH, photometric_validator),
+    ValidationTarget(Tag.PixelSpacing, ValidationLevel.VALID, Priority.LOW),
+    ValidationTarget(Tag.BitsStored, ValidationLevel.VALID, Priority.LOW, bits_validator),
+    ValidationTarget(Tag.PartialView, ValidationLevel.PRESENT, Priority.LOW),
+    ValidationTarget(Tag.LossyImageCompression, ValidationLevel.VALID, Priority.LOW),
+    ValidationTarget(Tag.Density, ValidationLevel.VALID, Priority.LOW),
+    ValidationTarget(Tag.WindowCenter, ValidationLevel.IGNORE, Priority.LOW),
+    ValidationTarget(Tag.WindowWidth, ValidationLevel.IGNORE, Priority.LOW),
+]
+
+
+class Validator:
+    def __init__(self, targets: Sequence[Union[ValidationTarget, Tuple[ValidationTarget, ...]]]):
+        self.targets = list(sorted(targets, reverse=True))
+
+    def __add__(self, other: "Validator") -> "Validator":
+        if not isinstance(other, Validator):
+            raise TypeError(f"Expected Validator for other, found {type(other)}")
+        return Validator(list(chain(self.targets, other.targets)))
+
+    def validate_dicom_object(self, dcm: Dicom) -> Sequence[ValidationResult]:
+        result: List[ValidationResult] = []
+        for target in self.targets:
+            if isinstance(target, Iterable):
+                all_tag_results = [t(dcm) for t in target]
+                tag_result = sum(all_tag_results, all_tag_results[0])
+            else:
+                tag_result = target(dcm)
+            result.append(tag_result)
+        return result
+
+    def validate_dicom_file(self, path: Path) -> Sequence[ValidationResult]:
+        if not path.is_file():
+            raise FileNotFoundError(path)
+        with pydicom.dcmread(path) as dcm:
+            result = self.validate_dicom_object(dcm)
+        return result
+
+    @classmethod
+    def default_validator(cls) -> "Validator":
+        return cls(DEFAULT_TARGETS)
+
+    @staticmethod
+    def _format(results: Sequence[ValidationResult], delim: str = "\t", color: bool = False) -> str:
+        lens = np.array([x.column_widths(color) for x in results])
+        lens = lens.max(axis=0)
+        lens += 2
```

Comment:
Could we assign a name to "2" so that the purpose of this line is clearer? I.e.
```python
CONST_XXXXX: Final[int] = 2
lens = lens.max(axis=0) + CONST_XXXXX
```